src/sentry.js

```diff
@@ -45,6 +51,82 @@ const preventNoise = (): void => {
   }
 };
 
+/**
+ * An HTTP request made by the app.
+ *
+ * One of the builtin "recognized breadcrumb types":
+ *   https://develop.sentry.dev/sdk/event-payloads/breadcrumbs/#breadcrumb-types
+ */
+type HttpBreadcrumb = $ReadOnly<{|
+  ...Breadcrumb,
+  type: 'http',
+  data: {|
+    url?: string,
+    method?: string,
+    status_code?: number,
+    reason?: string,
+  |},
+|}>;
+
+/**
+ * Scrubs possibly personal information from a breadcrumb.
+ *
+ * - Removes the realm hostnames in API requests, except for Kandra-hosted
+ *   realms.
+ */
+function scrubBreadcrumb(breadcrumb: Breadcrumb, hint?: BreadcrumbHint): Breadcrumb {
+  switch (breadcrumb.type) {
+    case 'http': {
+      // $FlowIgnore[incompatible-indexer]
+      // $FlowIgnore[incompatible-type]
+      const httpBreadcrumb: HttpBreadcrumb = breadcrumb;
+
+      const url = (() => {
+        const { url: urlStr } = httpBreadcrumb.data;
+        if (urlStr === undefined) {
+          return urlStr;
+        }
+        const parsedUrl = new URL(urlStr);
+        const accountStatuses = getAccountStatuses(store.getState());
+        if (
+          // prettier-ignore
+          (
+            // Most likely an API request to a Zulip realm. This will
+            // catch `/api/v1/server_settings` and `/api/v1/fetch_api_key` requests to a
+            // realm before it's added to the account-statuses state.
+            parsedUrl.pathname.startsWith('/api/v1')
+            // Definitely a request to a Zulip realm. Will catch requests to
+            // realms in the account-statuses state, including those without
+            // `/api/v1`, like `/avatar/{user_id}` (zulip/zulip@0f9970fd3
+            // confirms that this and potentially others don't / won't use
+            // `/api/v1`).
+            || accountStatuses.some(({ realm }) => parsedUrl.origin === realm.origin)
+          )
+          && !config.appOwnDomains.some(
+            domain => parsedUrl.host === domain || parsedUrl.host.endsWith(`.${domain}`),
+          )
```

Comment:
This is doing the same thing as in the other spot where we use `config.appOwnDomains`, right?

Let's pull this out as its own little named helper so that there's only one copy of it, because it is a bit tricky. Not sure there's a great place to group it with other things; so perhaps just its own tiny file `src/isAppOwnDomain.js`.